notebooks/find-quotes.py

```diff
@@ -0,0 +1,155 @@
+    # Convert quote data into page data
+    page_data = (
+        quotes_data.unique("page_text")
+        .select("FILE", "page_index", "page_text")
+        .sort("page_index")
+    )
+    page_data
+    return (page_data,)
+
+
+@app.cell
+def _(mo):
+    mo.md(r"""## View Results by Page""")
+    return
+
+
+@app.cell
+def _(mo, page_data):
+    page_slider = mo.ui.slider(start=1, stop=len(page_data.rows()), value=2)
+    return (page_slider,)
+
+
+@app.cell
+def _(mo, page_slider):
+    mo.vstack(
+        [
+            mo.md("Select Slider:"),
+            page_slider,
+            mo.md(f"Selected Page: {page_slider.value}"),
+        ]
+    )
+    return
+
+
+@app.cell
+def _():
+    # Store notes for pages (index by page id)
+    notes = {
+        18: """- Corresponds to page 25""",
+        38: """- Corresponds to page 43\n- Missing leading quote mark: "Frage an:\n<mark>**„**</mark>Bet gewissem"\n- Spurious quote mark: "die Naturprodukte nicht!<mark>**"**</mark> Hier"
+        """,
+        76: """- Corresponds to page 76\n- Missing ending quote mark, transcribed as period: "und Schön¬
+    geister<mark>**.**</mark>\n— heißt es\n- Missing ending quote mark: "die Verwirklichung der Philssophie<mark>**“**</mark> (a. a. O. S. 85)."
+    """,
+        256: """- Corresponds to page 231""",
+        301: """- Corresponds to page 268\n- Ending block quote not captured because it continues to the next page: "„Die Kommunisten [...]"
+        """,
+        499: """- Corresponds to page 289""",
+        545: """- Corresponds to page 493\n- Missing ending quote, transcribed as ¬: "die „Heilige Familie<mark>**¬**</mark> zurück"
+        """,
+        546: """- Corresponds to page 494\n- Spurious quote mark: "Produktionsprozesse\nleitet,<mark>**“**</mark> wiedergefunden"
+        """,
+        549: """- Corresponds to page 497\n- Potentially incorrect annotations. Does this page contain any direct quotations from Kapital, or are there just citations (e.g., the footnote)? - The text seems to be missing the closing quotation for a referenced title. As a result, several title mentions are grouped together.""",
+        661: """- Corresponds to page 612""",
+    }
+    return (notes,)
+
+
+@app.cell
+def _(find_quotes, mo, notes, page_data, page_slider):
+    row_id = page_slider.value - 1
+    page_id = page_data.row(row_id)[1]
+    page_text = page_data.row(row_id)[2]
+    results = [mo.md(f"## Page {page_id}")]
+    if page_id in notes:
+        results.append(mo.md(f"### Notes\n{notes[page_id]}"))
+    # Get matches
+    matches = find_quotes(page_text)
+    # Highlight (and bold) Page Text
+    highlighted_page = page_text
+    for match in sorted(matches, reverse=True, key=lambda x: x.span()):
+        start, end = match.span()
+        highlighted_page = (
+            highlighted_page[:start]
+            + "<mark>**"
+            + highlighted_page[start:end]
+            + "</mark>**"
+            + highlighted_page[end:]
```

Comment:
I'm not sure why, but the mark + bold in dark mode results in white text with yellow highlighting, which is unreadable. Suggest using only one of these (or pulling in the custom css for highlighting).

```suggestion
            highlighted_page[:start]
            + "<mark>"
            + highlighted_page[start:end]
            + "</mark>"
            + highlighted_page[end:]
```

Reply:
I guess this is a case of light vs. dark mode; for me the bold makes the text more legible within the highlight.

Reply:
Maybe a case for needing to create an issue to make the custom css highlighting more useful in light-mode and more generally accessible (i.e. colorblind friendly)

Reply:
I really just want our own "<highlight>" tag 😂